### ISSUE TITLE
Outlier detection ignores masks when in_memory is True

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8776]
_Issue [JP-3743](https://jira.stsci.edu/browse/JP-3743) was created on JIRA by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter):_

The introduction of ModelLibrary (JP-3690) introduced a bug where the drizzled array weights are ignored entirely when the `in_memory` parameter is set to True.  This can be seen by modifying [this unit test](https://github.com/spacetelescope/jwst/blob/bd7d2694504f77be9fef82c8d70e25f2204b2b00/jwst/outlier_detection/tests/test_outlier_detection.py#L598) as follows, such that it's meaningful with respect to the weights array and weight threshold:
```java
def test_create_median(three_sci_as_asn, tmp_cwd):
"""Test creation of median on disk vs in memory"""
    lib_on_disk = ModelLibrary(three_sci_as_asn, on_disk=True)
    lib_in_memory = ModelLibrary(three_sci_as_asn, on_disk=False)
    # make this test meaningful w.r.t. handling of weights
    with (lib_on_disk, lib_in_memory):
        for lib in [lib_on_disk, lib_in_memory]:
            for model in lib:
                model.wht = np.ones_like(model.data)
                model.wht[4,9] = 0.5
                lib.shelve(model, modify=True)

    median_on_disk = create_median(lib_on_disk, 0.7, on_disk=True)
    median_in_memory = create_median(lib_in_memory, 0.7, on_disk=False)

    assert np.isnan(median_in_memory[4,9])

    # Make sure the median library is the same for on-disk and in-memory
    assert np.allclose(median_on_disk, median_in_memory, equal_nan=True) ```

**Comment by stscijgbot-jp** [comment]
_Comment by [Ned Molter](https://jira.stsci.edu/secure/ViewProfile.jspa?name=emolter) on [JIRA](https://jira.stsci.edu/browse/JP-3743?focusedCommentId=718397#comment-718397):_

Fixed by #8777

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3743?focusedCommentId=718987#comment-718987):_

Internal issue, closing based on clean regression tests.
